docs/en/cookbook/generated-columns.rst

```diff
@@ -0,0 +1,99 @@
+Generated Columns
+=================
+
+Generated columns, sometimes also called virtual columns, are populated by
+the database engine itself. They are a tool for performance optimization, to
+avoid calculating a value on each query.
+
+You can define generated columns on entities and have doctrine map the values
+to your entity.
+
+Declaring a generated column
+----------------------------
+
+There is no explicit mapping instruction for generated columns. Instead, you
+specify that the column should not be written to, and define a custom column
+definition.
+
+.. code-block:: php
+
+    <?php
+
+    use Doctrine\ORM\Mapping as ORM;
+
+    #[ORM\Entity]
+    class Person
+    {
+        #[ORM\Column(type: "string")]
+        private $firstName;
```

Comment:
I think you could use typed properties here and below.

Reply:
oh, indeed. i think i copied from something else and did not pay attention :see_no_evil: